cfgov/cal/views.py

```diff
@@ -31,6 +26,32 @@
        PDFreactor = None
 
 
+class FilterErrorList(forms.utils.ErrorList):
+    def __str__(self):
+        return '\n'.join(str(e) for e in self)
+
+class FilterCheckboxList(forms.MultipleChoiceField):
+    def validate(self, value):
+        if self.required and not value:
+            msg = self.error_messages['required']
+            if self.label and '%s' in msg:
+                msg = msg % self.label
+            raise forms.ValidationError(msg, code='required')
+        return value
+
+class FilterDateField(forms.DateField):
+    def clean(self, value):
+        if value:
+            try:
+                get_date_string(value)
+            except ValueError:
+                msg = self.error_messages['invalid']
+                if '%s'in msg:
+                    msg = msg % value
+                raise forms.ValidationError(msg , code='invalid')
+
+        return super(FilterDateField, self).clean(value)
```

Comment:
Thanks, I understand now. That's a helpful clarification. I'll correct and also add unit tests here.